app/integrations/server/lib/eventArgumentHandler.ts

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { logger } from '../logger';
+import { settings } from '../../../settings/server';
+
+
+interface IKeyable {
+	[key: string]: any;
+}
+
+class EventArgumentHandler {
+	eventNameArgumentsToObject(...args: any): IKeyable {
+		const argObject: IKeyable = {
+			event: args[0],
+		};
+
+		switch (argObject.event) {
+			case 'sendMessage':
+				if (args.length >= 3) {
+					argObject.message = args[1];
+					argObject.room = args[2];
+				}
+				break;
+			case 'fileUploaded':
+				if (args.length >= 2) {
+					const arghhh = args[1];
+					argObject.user = arghhh.user;
+					argObject.room = arghhh.room;
+					argObject.message = arghhh.message;
+				}
+				break;
+			case 'roomArchived':
+				if (args.length >= 3) {
+					argObject.room = args[1];
+					argObject.user = args[2];
+				}
+				break;
+			case 'roomCreated':
+				if (args.length >= 3) {
+					argObject.owner = args[1];
+					argObject.room = args[2];
+				}
+				break;
+			case 'roomJoined':
+			case 'roomLeft':
+				if (args.length >= 3) {
+					argObject.user = args[1];
+					argObject.room = args[2];
+				}
+				break;
+			case 'userCreated':
+				if (args.length >= 2) {
+					argObject.user = args[1];
+				}
+				break;
+			default:
+				logger.outgoing.warn(`An Unhandled Trigger Event was called: ${ argObject.event }`);
+				argObject.event = undefined;
+				break;
+		}
+
+		logger.outgoing.debug(`Got the event arguments for the event: ${ argObject.event }`, argObject);
```

Comment:
Since you assigned `undefined` to `argObject.event` on L57, this line will print `undefined` instead of the actual received type. Probably, you can move this log to the top of the function?

Reply:
Nice observation, I will move it to the top of the function